mrp_bom_comparison/wizards/mrp_bom_comparison.py

```diff
@@ -0,0 +1,247 @@
+        # Get all data for each BoM
+        bom1_data = self._get_all_data(self.bom1_id)
+        bom2_data = self._get_all_data(self.bom2_id)
+        # Make the comparison between them
+        diff = DictDiffer(bom2_data, bom1_data)
+        # Iterate over data to generate lines to display on the report
+        for p_id in diff.changed():
+            v1 = bom1_data[p_id]
+            v2 = bom2_data[p_id]
+            vals = {
+                'wiz_id': self.id,
+                'product_id': p_id,
+                'bom1_qty': v1['bom_qty'],
+                'bom2_qty': v2['bom_qty'],
+                'diff_qty': v2['bom_qty'] - v1['bom_qty'],
+                'state': 'changed',
+            }
+            _logger.info(
+                u"\tProduct updated: %s (ID=%s) %s -> %s",
+                v1['product_code'], p_id,
+                v1['bom_qty'], v2['bom_qty'])
+            comparison_line_model.create(vals)
+        for p_id in diff.added():
+            v2 = bom2_data[p_id]
+            vals = {
+                'wiz_id': self.id,
+                'product_id': p_id,
+                'bom1_qty': 0.0,
+                'bom2_qty': v2['bom_qty'],
+                'diff_qty': v2['bom_qty'],
+                'state': 'added',
+            }
+            _logger.info(
+                u"\tProduct added: %s (ID=%s) -> %s",
+                v2['product_code'], p_id, vals['diff_qty'])
+            comparison_line_model.create(vals)
+        for p_id in diff.removed():
+            v1 = bom1_data[p_id]
+            vals = {
+                'wiz_id': self.id,
+                'product_id': p_id,
+                'bom1_qty': v1['bom_qty'],
+                'bom2_qty': 0.0,
+                'diff_qty': -v1['bom_qty'],
+                'state': 'removed',
+            }
+            _logger.info(
+                u"\tProduct removed: %s (ID=%s) -> %s",
+                v1['product_code'], p_id, vals['diff_qty'])
+            comparison_line_model.create(vals)
+        _logger.info(
+            u"BoM comparison between '%s' and '%s': printing report...",
+            self.bom1_id.product_tmpl_id.default_code,
+            self.bom2_id.product_tmpl_id.default_code)
+        # Return the report
+        return self.env['report'].get_action(
+            self, 'mrp_bom_comparison.report_mrp_bom_comparison')
+
+
+class WizardMrpBomComparisonLine(models.TransientModel):
+    _name = 'wizard.mrp.bom.comparison.line'
+    _description = "BoM line difference"
+
+    wiz_id = fields.Many2one('wizard.mrp.bom.comparison', u"Wizard")
+    product_id = fields.Many2one('product.product', u"Product")
+    bom1_qty = fields.Float(
+        u"v1-Qté", digits=dp.get_precision('Product Unit of Measure'))
+    bom2_qty = fields.Float(
+        u"v2-Qté", digits=dp.get_precision('Product Unit of Measure'))
+    diff_qty = fields.Float(
+        u"Écart qté", digits=dp.get_precision('Product Unit of Measure'))
```

Comment:
You should put this in English and then add translations.

Reply:
Thank you! This is a old module written for 6.0 a long time ago, I hope all terms are fixed now.